nilearn/datasets/tests/test_func.py

```diff
@@ -25,13 +24,70 @@
 
 
 @pytest.fixture()
-def request_mock():
+def request_mocker():
     tst.setup_mock(utils, func)
     yield
     tst.teardown_mock(utils, func)
 
 
-def test_fetch_haxby(tmp_path, request_mock):
```

Comment:
I moved the code block, there was a bunch of stuff, globals, non-test definitions sitting in the middle of all the tests and so I moved it. It has made the PR a little harder to review, but I didn't resist, it was looking terrible.

The transplanted code block is further down, no actual code change.